In [1]:
import pandas as pd
 
 
url = "https://raw.githubusercontent.com/mricardo89/data-mining/main/Unit-II/datasets/titanic.csv"
df = pd.read_csv(url)

A - Tasa de sobrevivencia

In [3]:
df['Survived'].value_counts(normalize=True)

Survived
0    0.616162
1    0.383838
Name: proportion, dtype: float64

Pregunta A:

--> Sobrevivió el 38.3838% de los pasajeros del barco. Más de la mitad (61.6162%) no sobrevivió.

B - Sobrevivencia en Mujeres/Hombres

In [9]:
sobrevivientes = df.groupby('Sex')['Survived'].mean()
print(sobrevivientes)


Sex
female    0.742038
male      0.188908
Name: Survived, dtype: float64


Pregunta B: 

--> Sobrevivió el 74.2% de las mujeres vs. un 18.89% de hombres
Se puede decir que el código se respetó.

C - Cuartiles en Tarifas

In [11]:
df['Fare'].quantile([0.25, 0.75])

0.25     7.9104
0.75    31.0000
Name: Fare, dtype: float64

Calculando IQR y rango intercuartilico

In [ ]:
q1 = df['Fare'].quantile(0.25)
q3 = df['Fare'].quantile(0.75)
iqr = q3-q1
print("IQR =", iqr)

lim_sup = q3 + 1.5 * iqr
print("Limite superior aceptable =", lim_sup)

IQR = 23.0896
Limite superior aceptable = 65.6344


In [16]:
outliers = df[df['Fare']>lim_sup]
print("Cantidad de pasajeros que pagaron por encima del limite superior aceptable:", len(outliers))

Cantidad de pasajeros que pagaron por encima del limite superior aceptable: 116


In [18]:
print(outliers['Pclass'].value_counts())

Pclass
1    104
3      7
2      5
Name: count, dtype: int64
